internal/store/builder.go

```diff
@@ -541,6 +561,46 @@ func (b *Builder) buildStores(
 	return stores
 }
 
+func (b *Builder) buildMetadataOnlyStores(
```

Comment:
I might be missing something, but it seems you could call `buildStores` right away since this is an exact copy of its code.

Going even further, maybe we don't even need `buildMetadataOnlyStores` and could just call `buildStores` directly when building resource specific stores like:

```diff
-	return b.buildStoresFunc(configMapMetricFamilies(b.allowAnnotationsList["configmaps"], b.allowLabelsList["configmaps"]), &v1.ConfigMap{}, createConfigMapListWatch, b.useAPIServerCache)
+	return b.buildStoresFunc(configMapMetricFamilies(b.allowAnnotationsList["configmaps"], b.allowLabelsList["configmaps"]), &metav1.PartialObjectMetadata{}, createConfigMapListWatch, b.useAPIServerCache)
```

Reply:
The reason I created that is the different signature of the listWatchFunc (using metadata.Interface not clientset.Interface). Is there's a different way to do that, I'm all ears :)